server/src/core/user/user-service.ts

```diff
@@ -228,12 +235,25 @@ export class Users implements UserService {
   async login(args: UserLoginArgs): Promise<UserLoginResult> {
     validators.validatesLogin(args);
     try {
-      const user = await this.collection.findOne({ phone: args.phone });
-      if (!user) throw createLoginError();
-
-      const passwordCorrect = await verifyPassword(user.password, args.password);
-      if (!passwordCorrect) {
-        throw createLoginError();
+      let user;
+      if (args.phone) {
+        user = await this.collection.findOne({ phone: args.phone });
+        if (!user) throw createLoginError();
+
+        if (args.password) {
+          const passwordCorrect = await verifyPassword(user.password, args.password);
+          if (!passwordCorrect) throw createLoginError();
+        }
+        else if (args.googleIdToken) {
```

Comment:
Why would you need to log in with both your phone number and google account id? Isn't either one of them enough on its own? As in, if you're logging in with your phone, you don't need to provide your google id token (In fact, you wouldn't even have one in the first place). Conversely, if you're logging in using your Google account, you need only your google id token and nothing more. I don't know, maybe there's something I'm missing, and perhaps you could point it out.

Reply:
Yeah, that's what I initially did but it was a bug since when you create an account, the user is immediately logged in with the same args. That's why I added this case to account for that.

Reply:
This is for the case where you create your account. It automatically signs you in with the same args. That's why I added this case.